test/runnable/test_cdcmp.d

```diff
@@ -0,0 +1,1477 @@
+            {
+                foreach (op; ops)
+                {
+                    foreach (suffix; [tuple("zero", "Zero!"~type.stringof), tuple(type.stringof, type.stringof)])
+                    {
+                        args = ["objdump", "--disassemble", "--disassembler-options=intel-mnemonic",
+                            "--section=.text.testee_" ~ type.stringof ~ "_" ~ opName(op) ~ "_" ~ suffix[0],
+                            __FILE__.baseName.setExtension(".o")];
+                        auto p = pipeProcess(args);
+                        formattedWrite(sink, "    Code!(%s, \"%s\", %s)([\n", type.stringof, op, suffix[1]);
+                        foreach (line; p.stdout.byLine.find!(ln => ln.matchFirst(ctRegex!">:$"))
+                            .dropOne.until!(ln => ln.canFind("...")))
+                        {
+                            replaceFirstInto!formatASM(sink, line, asmRE);
+                        }
+                        formattedWrite(sink, "    ]),\n");
+                        enforce(wait(p.pid) == 0, p.stderr.byLine.join("\n"));
+                    }
+                }
+            }
+            formattedWrite(sink, ");\n\n");
+        }
+        {
+            auto content = src.readText;
+            auto f = File(src, "w");
+            auto orng = f.lockingTextWriter;
+            immutable string start = "// dfmt off";
+            immutable string end = "// dfmt on";
+            replaceFirstInto!((_, orng) => formattedWrite(orng, start ~ "\n%s" ~ end, sink.data))(orng,
+                    content, ctRegex!(`^` ~ start ~ `[^$]*` ~ end ~ `$`, "m"));
+        }
+    }
+}
+// test code
+else:
+
+template testee(T, string op, T2)
+{
+    static if (is(T2 == Zero!U, U))
+        enum mangling = "testee_" ~ T.stringof ~ "_" ~ opName(op) ~ "_" ~ "zero";
+    else
+        enum mangling = "testee_" ~ T.stringof ~ "_" ~ opName(op) ~ "_" ~ T2.stringof;
+    pragma(mangle, mangling)
+    bool testee(T a, T2 b)
+    {
+        return mixin("a " ~ op ~ " b");
+    }
+}
+
+struct Zero(T)
+{
+    enum T zero = 0;
+    static alias zero this;
+}
+
+// holding the expected byte sequence
+struct Code(T_, string op_, T2_)
+{
+    alias T = T_;
+    alias op = op_;
+    alias T2 = T2_;
+    ubyte[] code;
+}
+
+alias AliasSeq(Args...) = Args;
+
+// dfmt off
+alias baselineCases = AliasSeq!(
+    Code!(ubyte, "<", Zero!ubyte)([
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* xor    eax,eax                 */ 0x31, 0xc0,
+        /* pop    rbp                     */ 0x5d,
+        /* ret                            */ 0xc3,
+    ]),
+    Code!(ubyte, "<", ubyte)([
+        /* push   rbp                     */ 0x55,
+        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
+        /* sub    rsp,0x10                */ 0x48, 0x83, 0xec, 0x10,
```

Comment:
All the 0xNN data can get incredibly tedious to create. In the future, use hexStrings -
```
0x48, 0x83, 0xEC, 0x10,
```
becomes:
```
hexString!"48 83 EC 10"
```

Reply:
This file is auto-generated ;-)

Reply:
HexString relies on phobos, so I avoided that.